pkg/metrics/iterable_metrics.go

```diff
@@ -114,3 +115,32 @@ func StartIteration(iterableSeries *IterableSeries, sink func(SerieSink), source
 	iterableSeries.senderStopped()
 	<-done
 }
+
+// StartSerialization starts the serialization for series and sketches.
+// `sink` callback is responsible for adding the data. It runs in the current goroutine.
+// `serieSource` callback is responsible for consuming the series. It runs in its OWN goroutine.
+// `sketchesSource` callback is responsible for consuming the sketches. It runs in its OWN goroutine.
```

Comment:
These arguments seem to be named opposite of what they do: to me `sink` sounds more like a destination or consumer, and `source` is an origin or producer.

Reply:
`sink` and `source` were named based on their usage. 
A user of iterable metric sends metrics to a sink and gets metrics from a source which makes sense in my opinion. 

`pkg/metrics/iterable_metrics.go` is the implementation and so the logic is the opposite. A user sends data to a `sink` and the `sink` is read by the functions in `iterable_metrics.go`.

I don't have a strong opinion on keeping these namings. Let me know if you have a better names in mind.
 
Note that most of this code was added in a previous PR.